docs/13_state_management_investigation.md

```diff
@@ -0,0 +1,1596 @@
+
+#### Easy to learn
+
+Score: 1
+
+Redux is hard to learn, because of its key concept, which is not clear to newcomers with this state management. Too many different parts that you need to understand: `store`, `actions`, `middleware`, `reducers`.
+
+The package has the same concept as `redux.js`, so if you have experience working with it, `flutter_redux` will be understandable for you.
+
+Namings in Redux is not intuitive, and at first it is hard to understand what middleware or reducers are used for.
+
+#### Centralized analytics
+
+Score: 5
+
+With Redux we can use a middleware, that can store information about our state and actions, that dispatched across whole app. As this is a common task, there is a [redux_logging](https://pub.dev/packages/redux_logging) package that prints the latest action & state. So, with that we can easily provide centralized analytics.
+
+#### State immutability
+
+Score: 5
+
+The application `state` in Redux is immutable and it is a core concept. The only way to mutate the `state` is to `dispatch` an `action`, that has a new `state` as a result.
+
+#### Ability to use outside Flutter
+
+Score: 5
+
+The [redux](https://pub.dev/packages/redux) package developed to work with `Dart`. So we can use Redux outside of `flutter`.
+
+To work with `flutter` there is a [flutter_redux](https://pub.dev/packages/flutter_redux) package, so we have a good flutter integration.
+
+Also, business logic is separated from the UI and split into `actions`, `middlewares`, `reducers`, so we can easily reuse the code.
+
+#### Pros
+
+1. [Asynchronous](#Asynchronous-3) is a part of Redux, but need to create separate functions to do some asynchronous stuff.
+2. It has a high level of [maintainability](#Maintainability-3).
+3. You can make [state snapshots](#State-snapshot-3).
+4. Easy to [debug](#Debugging-3).
+5. Support [Undo and redo](#Undo-and-redo-actions-3) actions.
+6. Good [testability](#Testability-3).
+7. [State immutability](#State-immutability-3) is a core concept.
+8. [Ability to use outside Flutter](#Ability-to-use-outside-Flutter-3)
+9. [Centralized analytics](#Centralized-analytics-3)
+10. Store changes are perfectly predictable.
+11. Unidirectional data flow.
+
+### Cons
+
+1. It has a support of [reactivity](#Reactivity-3) but needs to add a separate package.
+2. It requires a lot of [boilerplate](#boilerplate-absence-3).
+3. [High learning curve](#Easy-to-learn-3).
+4. The store will get very large with large applications.
+5. Every widget can access to the global store.
+
+## [Provider](https://pub.dev/packages/provider)
+
+The provider is the state management mechanism based on the flutter `InheritedWidget`, that can expose any kind of state object, including BLoC, streams, futures, and others. Because of its simplicity and flexibility, Google announced at Google I/O ’19 that Provider is now its preferred package for state management.
```

Comment:
```suggestion
The provider is the state management mechanism based on the flutter `InheritedWidget`, that can expose any kind of state object, including BLoC, streams, futures, and others. Because of its simplicity and flexibility, Google announced at Google I/O ’19 that Provider is now a preferred package for state management.
```